rpcs3/Emu/GDB.cpp

```diff
@@ -79,32 +76,32 @@ bool check_errno_again()
 }
 
 std::string u32_to_hex(u32 i) {
-	return fmt::format("%" HEX_U32, i);
+	return fmt::format("%" PRIx32, i);
 }
 
 std::string u64_to_padded_hex(u64 value) {
-	return fmt::format("%.16" HEX_U64, value);
+	return fmt::format("%.16" PRIx64, value);
 }
 
 std::string u32_to_padded_hex(u32 value) {
-	return fmt::format("%.8" HEX_U32, value);
+	return fmt::format("%.8" PRIx32, value);
 }
 
 u8 hex_to_u8(std::string val) {
 	u8 result;
-	sscanf_s(val.c_str(), "%02hhX", &result);
+	sscanf_s(val.c_str(), "%02" PRIX8, &result);
```

Comment:
std::from_chars seems better suited performance wise + able to etror check if the input is not a valid hexademial integer representation as string. I also think these functions can be declared using one template function, then using std::bind. Last but not least, std::string_view should be used instead of std::string.

Reply:
Seems reasonable. I haven't used std::bind before - what would this look like?

Reply:
`constexpr auto hex_to_u8 = std::bind(hex_to<u8>);` for example.

Reply:
Or actually `constexpr auto& hex_to_u8 = hex_to<u8>;` means the same.

Reply:
I went with the latter. Thanks for the suggestion!